daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

```diff
@@ -0,0 +1,59 @@
+---
+type: docs
+title: "Wasm middleware"
+linkTitle: "Wasm"
+description: "Test your HTTP pipeline is functioning with the wasm middleware"
+aliases:
+- /developing-applications/middleware/supported-middleware/middleware-wasm/
+---
+
+The wasm [HTTP middleware]({{< ref middleware.md >}}) use a WASM file to handle request and response.
```

Comment:
```suggestion
The Wasm [HTTP middleware]({{< ref middleware.md >}}) component enables you to use a Wasm file to handle requests and responses.
```

Reply:
Can you add more information here on 
1) Why this is useful to do? Why use Wasm?
2) Give 2-3 examples of how this can be used. When would someone use this
3) How do they need to write the Wasm file/component? Are there any special entry points and what do I need to expose in the Wasm component? This may be as simple as pointing to a good resource on using Wasm, and at least give people a direction to look
4) Best of all would you consider add sample code into the Dapr Samples repo? This would really help people with a code exampl.

Reply:
OK, I will give some examples to make it clear.

Reply:
Thanks!